src/tiled/objectsdock.cpp

```diff
@@ -401,6 +405,45 @@ void ObjectsView::selectedObjectsChanged()
     }
 }
 
+void ObjectsView::setColumnVisibility(bool visible)
+{
+    QAction *action = qobject_cast<QAction*>(sender());
+    if (action)
+        header()->setSectionHidden(action->data().toInt(), !visible);
+}
+
+void ObjectsView::showCustomMenu(const QPoint &point)
+{
+    Q_UNUSED(point)
+    QMenu contextMenu(this);
+    contextMenu.addActions(mActions);
+    contextMenu.exec(QCursor::pos());
+}
+
+void ObjectsView::updateColumnVisibilityActions()
+{
+    hideExtraColumns();
+    mActions.clear();
+    QAbstractItemModel *model = mProxyModel->sourceModel();
+    for (int i = 0; i < model->columnCount(); i++) {
+        if (i == MapObjectModel::Name || i == MapObjectModel::Type)
```

Comment:
I think it's fine to allow hiding the Type column, it just needs to stay visible by default.